src/Sulu/Bundle/CustomUrlBundle/EventListener/UpdateLocalizationListener.php

```diff
@@ -0,0 +1,45 @@
+<?php
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\CustomUrlBundle\EventListener;
+
+use Sulu\Component\Webspace\Analyzer\RequestAnalyzerInterface;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+
+/**
+ * Update localization in case of custom-url route.
+ */
+class UpdateLocalizationListener
+{
+    /**
+     * @var RequestAnalyzerInterface
+     */
+    private $requestAnalyzer;
+
+    public function __construct(RequestAnalyzerInterface $requestAnalyzer)
+    {
+        $this->requestAnalyzer = $requestAnalyzer;
+    }
+
+    /**
+     * Update locale in request analyzer.
+     *
+     * @param GetResponseEvent $event
+     */
+    public function onKernelRequest(GetResponseEvent $event)
+    {
+        $request = $event->getRequest();
+        if ($request->get('_custom_url') === null) {
+            return;
+        }
+
+        $this->requestAnalyzer->updateLocale($request->get('_custom_url')->getTargetLocale(), $request);
```

Comment:
And we actually intended that the values of the `RequestAnalyzer` are immutable, this might possibly introduce errors quite hard to find. I think it would be better to include that logic in `RequestAnalyzer::analyze` somehow (probably not directly, because this information is coming from another bundle, but via some `Strategy` or a similar concept).